Add HostPath example

pohly opened the pull request:
For someone trying to get started with CSI the existing documentation
stops right at the point where someone is ready to deploy a CSI driver
and experiment with it.

The new example is based on the .yaml files provided by Serguei
Bezverkhi on slack. The troubleshooting section addresses some
real-world issues encountered when trying to get CSI working (local
error with incomplete cluster config and out-dated hostpath image).

Note to reviewer:
the absolute links to the .yaml file do not work yet - the PR first has to be merged.
It is debatable whether those URL should point to master or the current copy
of the documentation on kubernetes-csi.github.io. I picked latest master because
the images are also getting updated independently from the documentation.

sbezverk commented on book/src/Example.md:
````diff
@@ -0,0 +1,95 @@
+```
+kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/docs/master/book/src/example/csi-pvc.yaml
+```
+
````
Do you think you could add the output of ```kubectl get pv```console? I think it will help a person who uses this guide to validate locally observed results.

pohly replied:
The output is already present, under "Troubleshooting". Is that too late because users won't read that far?

I can also interleave it with the commands here.

sbezverk replied:
I think adding the output command here will help a user to validate results he/she observing locally.  Troubleshooting section is great, but the user needs an indicator.  Comparing what you provide in the document with what he/she sees locally is good indicator and if something goes wrong then they can refer to the troubleshooting section.

pohly replied:
Okay, I'll make the usage section a bit more "interactive" by also including output.

pohly replied:
Done. I made the "Usage" section a mixture of commands+output, so one can see immediately what the result should. The "Troubleshooting" section now focuses on specific problems or aspects and does not show anymore what a working cluster looks like, because that already came earlier.

I also added something about VolumeAttachment.

xing-yang commented on book/src/Example.md:
````diff
@@ -0,0 +1,95 @@
+team](https://github.com/kubernetes/community/tree/master/sig-storage) and/or
+[run the e2e test](https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md#local-clusters):
+```
+go run hack/e2e.go -- --provider=local --test --test_args="--ginkgo.focus=Feature:CSI"
````
hack/e2e.sh

pohly replied:
No, "go run" is really invoked for hack/e2e.go. That's a Go wrapper program which then invokes e2e.sh. See https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md#building-kubernetes-and-running-the-tests

xing-yang replied:
Got it. Thanks!

lpabon commented:
when the conflicts are resolved, I'll merge.

pohly commented:
On Sat, 2018-03-10 at 04:42 +0000, Luis Pabón wrote:
> when the conflicts are resolved, I'll merge.

Rebased.

I also added .gitignore for docs/example, which is a copy of the .yaml
snippets and not needed for anything.

pohly commented:
@lpabon I rebased, this should be ready for merging now.
